wagtail/admin/tests/test_views_generic.py

```diff
@@ -58,19 +58,29 @@ def test_non_url_safe_primary_key(self):
             response, "non-url-safe pk modelwithstringtypeprimarykey model"
         )
 
-    def test_using_quote_in_edit_url(self):
-        object_pk = 'string-pk-:#?;@&=+$,"[]<>%'
+    def test_unquote_sensitive_primary_key(self):
+        object_pk = "web_407269_1"
         response = self.get(quote(object_pk))
-        edit_url = response.context_data["action_url"]
-        edit_url_pk = edit_url.split("/")[-2]
-        self.assertEqual(edit_url_pk, quote(object_pk))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(
+            response, "unquote-sensitive modelwithstringtypeprimarykey model"
+        )
```

Comment:
The bug only affects snippets, and not the generic views more generally, as only the snippet views call `get_object` multiple times. However, as the fix is in the generic view it seemed sensible to add test coverage at both levels.